### DISCUSSION TITLE
Are rows internally sorted by primary-key?

### DISCUSSION THREAD

**Discussion by Sebastian-Nielsen** [planetscale/discussion#584] [FAQs]
Are rows internally sorted by primary-key?
What I am trying to ask is: Can you rely on rows being sorted by primary-key if you don't specify an 'order by' assuming you are executing a simple query by which I mean a query that only makes use of  `SELECT, FROM, WHERE, IN`?

**Comment by orware** [comment]
Hello @Sebastian-Nielsen,

In a case like this the details of the SQL standard itself will probably be best to rely on as shared by a user in the following Stack Overflow reply:
https://dba.stackexchange.com/a/6053

In practice, I have observed that by default my queries will tend to be ordered by the auto incrementing primary key value I typically have defined for my tables, in ascending order, but if that cannot be truly relied upon consistently as shared in the answer above you will want to ensure the ordering you desire by applying the use of an `ORDER BY` clause within your query.
